src/main/scala/org/broadinstitute/dsde/firecloud/service/NamespaceService.scala

```diff
@@ -0,0 +1,74 @@
+package org.broadinstitute.dsde.firecloud.service
+
+import akka.actor.{Actor, Props}
+import akka.pattern._
+import org.broadinstitute.dsde.firecloud.{Application, FireCloudExceptionWithErrorReport}
+import org.broadinstitute.dsde.firecloud.core.AgoraPermissionHandler
+import org.broadinstitute.dsde.firecloud.dataaccess.AgoraDAO
+import org.broadinstitute.dsde.firecloud.model.MethodRepository.{AgoraPermission, FireCloudPermission}
+import org.broadinstitute.dsde.firecloud.model.{RequestCompleteWithErrorReport, UserInfo}
+import org.broadinstitute.dsde.firecloud.service.NamespaceService.{GetPermissions, PostPermissions}
+import org.broadinstitute.dsde.firecloud.service.PerRequest.{PerRequestMessage, RequestComplete}
+import spray.http.StatusCodes._
+import spray.httpx.SprayJsonSupport
+import spray.httpx.SprayJsonSupport._
+import spray.json.DefaultJsonProtocol._
+import spray.json._
+import org.broadinstitute.dsde.firecloud.model.ModelJsonProtocol._
+
+import scala.concurrent.{ExecutionContext, Future}
+
+object NamespaceService {
+  case class GetPermissions(ns: String, entity: String)
```

Comment:
Probably a good idea. That lets the compiler know if some cases are not handled within a match block.